app/src/main/java/com/github/brandon_patterson/chemist/matter/Element.java

```diff
@@ -0,0 +1,192 @@
+    CS(R.string.element_name_055, 55),
+    BA(R.string.element_name_056, 56),
+    LA(R.string.element_name_057, 57),
+    CE(R.string.element_name_058, 58),
+    PR(R.string.element_name_059, 59),
+    ND(R.string.element_name_060, 60),
+    PM(R.string.element_name_061, 61),
+    SM(R.string.element_name_062, 62),
+    EU(R.string.element_name_063, 63),
+    GD(R.string.element_name_064, 64),
+    TB(R.string.element_name_065, 65),
+    DY(R.string.element_name_066, 66),
+    HO(R.string.element_name_067, 67),
+    ER(R.string.element_name_068, 68),
+    TM(R.string.element_name_069, 69),
+    YB(R.string.element_name_070, 70),
+    LU(R.string.element_name_071, 71),
+    HF(R.string.element_name_072, 72),
+    TA(R.string.element_name_073, 73),
+    W(R.string.element_name_074, 74),
+    RE(R.string.element_name_075, 75),
+    OS(R.string.element_name_076, 76),
+    IR(R.string.element_name_077, 77),
+    PT(R.string.element_name_078, 78),
+    AU(R.string.element_name_079, 79),
+    HG(R.string.element_name_080, 80),
+    TL(R.string.element_name_081, 81),
+    PB(R.string.element_name_082, 82),
+    BI(R.string.element_name_083, 83),
+    PO(R.string.element_name_084, 84),
+    AT(R.string.element_name_085, 85),
+    RN(R.string.element_name_086, 86),
+    FR(R.string.element_name_087, 87),
+    RA(R.string.element_name_088, 88),
+    AC(R.string.element_name_089, 89),
+    TH(R.string.element_name_090, 90),
+    PA(R.string.element_name_091, 91),
+    U(R.string.element_name_092, 92),
+    NP(R.string.element_name_093, 93),
+    PU(R.string.element_name_094, 94),
+    AM(R.string.element_name_095, 95),
+    CM(R.string.element_name_096, 96),
+    BK(R.string.element_name_097, 97),
+    CF(R.string.element_name_098, 98),
+    ES(R.string.element_name_099, 99),
+    FM(R.string.element_name_100, 100),
+    MD(R.string.element_name_101, 101),
+    NO(R.string.element_name_102, 102),
+    LR(R.string.element_name_103, 103),
+    RF(R.string.element_name_104, 104),
+    DB(R.string.element_name_105, 105),
+    SG(R.string.element_name_106, 106),
+    BH(R.string.element_name_107, 107),
+    HS(R.string.element_name_108, 108),
+    MT(R.string.element_name_109, 109),
+    DS(R.string.element_name_110, 110),
+    RG(R.string.element_name_111, 111),
+    CN(R.string.element_name_112, 112),
+    NH(R.string.element_name_113, 113),
+    FL(R.string.element_name_114, 114),
+    MC(R.string.element_name_115, 115),
+    LV(R.string.element_name_116, 116),
+    TS(R.string.element_name_117, 117),
+    OG(R.string.element_name_118, 118);
+
+    private int name_id;
+    private int atomicNumber;
+
+    Element(int name_id, int atomicNumber)
+    {
+        this.name_id = name_id;
+        this.atomicNumber = atomicNumber;
+    }
+
+    public int getAtomicNumber()
+    {
+        return atomicNumber;
+    }
+
+    public String getName(Context context)
+    {
+        return context.getResources().getString(name_id);
```

Comment:
What is `Context`? I can't seem to find any concrete classes for this. How is the name gathered? Does this simply go the the elements XML?

Reply:
So getting to `Context` was a bit of a journey. To paraphrase:

- I tried to access resources the same way we would in an `Activity`, but that didn't work. (I think activities have info built-into them that plain classes don't have access to.)
- I searched for solutions, and there was one that could work without needing to pass an extra parameter, but it turns out that only works for System-level resources (and string resources are app-level).
- I found the `Context` approach next (see [SO](https://stackoverflow.com/questions/7666589/using-getresources-in-non-activity-class))

Context.getResources() effectively accesses the xml elements directly. (They've technically been turned into auto-generated java files, but you can mostly ignore that detail.) There's probably other information, like country/language settings, but that's also not important for us here.

It looks like an `Activity` class is actually a subclass of `Context`, so an activity can just pass itself (`this`) into the function if it ever needs to retrieve a name, so it should be fairly easy to work with in practice.

Having `Context` parameter also makes testing way more convenient, since I don't think unit tests have access to the resource files. The ability to pass a fake (mocked) context to the function isolates the component that we're testing.